packages/bitcore-node/src/routes/api/block.ts

```diff
@@ -52,6 +53,34 @@ router.get('/:blockId', async function(req: Request, res: Response) {
   }
 });
 
+router.get('at-time/:time', async function(req: Request, res: Response) {
+  let { time, chain, network } = req.params;
+  try {
+    const [block] = await BlockStorage.collection
+      .find({
+        $query: {
+          chain,
+          network,
+          timeNormalized: { $lte: new Date(time) }
+        }
+      })
+      .limit(1)
+      .sort({ timeNormalized: -1 })
```

Comment:
Is timeNormalized an index already?

Reply:
Yeah, `src/models/block.ts` 

`{chain: 1, network: 1, timeNormalized: 1}`